### ISSUE TITLE
Complication not Updating in Modular Ultra

### ISSUE THREAD

**Issue by nectario** [nightscout/nightguard#257]
It seems the complication never updates. Even one hour later, and it never updated. The Apple face I'm using is the Modular Ultra. I have an Apple Watch Ultra 2.

**Comment by dhermanns** [comment]
Hi! You would have to configure Background Updates:

https://h3rmanns.medium.com/enabling-background-updates-for-you-iphone-and-watch-4b8167a70b24

If that is configured correctly, what complication are you using?

The gauge complication should work fine. The rectangular one seems to need a fix on watchos 10.

**Comment by nectario** [comment]
I have already done this. Nothing works. Complication will not update. Sometimes an entire hour will pass without updating. I am using the Modular Ultra face...

**Comment by nectario** [comment]
Is this an issue with the complication or Apple Watch? It seems other complications are not updating either...

**Comment by dhermanns** [comment]
The circular nightguard complications are updating fine for me. The rectangular seems to have a problem, but just on the watch. And only after updating to watchos10.

**Comment by nectario** [comment]
How often is this supposed to update? I now see it updates every 20 minutes...

**Comment by dhermanns** [comment]
That is already very good. You can try to improve it e.g. by placing multiple Complications on your watch and ios widgets.

Each widget is getting an own update budget from Apple.

**Comment by nectario** [comment]
Thanks. Nothing seems to work. I tested again and tried to measure how long it will update. So, more than an hour later, it never updated...

**Comment by nectario** [comment]
![2023-10-31 19 12 54](https://github.com/nightscout/nightguard/assets/22030060/12bf4ed4-033d-4c4d-8f6e-c908a01d3d99)
Here is a screenshot

**Comment by dhermanns** [comment]
Try the latest TestFlight and the rectangular complication. It should update much better.

**Comment by grahamjenson** [comment]
I have the same issues with the updates not coming through on Apple Watch Ultra 2, complications in Modular ultra. It updates immediately if I click through to the app though.

**Comment by dhermanns** [comment]
Next Build 842 will be available soon. All complications should update mich more frequently.

Till now, I was using background updates to update the complications. These are very infrequent. Now the TimelineProvider, used for widgets, updates it on its own. That seems to work MUCH better.

You should see this new behavior in the rectangular complication in build 841 already.

**Comment by nectario** [comment]
Wow, thanks for addressing this issue! Looking forward to trying it! It seems this issue to be impacting A LOT of other complications as well. We are talking about no update even after an hour (all of them) except the weather one and compass....

**Comment by grahamjenson** [comment]
Awesome, thank you. Looking forward to the update.

**Comment by keesvanlierop** [comment]
Any idea when this will become generally available? Looking forward!

**Comment by dhermanns** [comment]
I'm currently rewriting background updates and therefore I'm planning to release ever as a bigger release.

But you can use the TestFlight till then...
